# Revisão de código 

Olá! 

Meu nome é Suelen. Estou feliz em revisar seu projeto hoje!

Quando vejo um erro pela primeira vez, apenas aponto. Deixarei você encontrá-lo e corrigi-lo sozinho. Além disso, ao longo do texto, farei algumas observações sobre melhorias no código e também farei comentários sobre suas percepções sobre o assunto. Mas se você ainda não consegue lidar com essa tarefa, darei uma dica mais precisa na próxima iteração e também alguns exemplos práticos. Estarei aberto a feedbacks e discussões sobre o tema.

Você pode encontrar meus comentários em caixas verdes, amarelas ou vermelhas como estas:


<div class="alert alert-block alert-success">
<b>Comentário: </b> <a class="tocSkip"></a>

Sucesso. Tudo está correto.
</div>

<div class="alert alert-block alert-warning">
<b>Comentário: </b> <a class="tocSkip"></a>

Observações. Algumas recomendações.
</div>

<div class="alert alert-block alert-danger">

<b>Comentário: </b> <a class="tocSkip"></a>

O bloco requer algumas correções. O trabalho não pode ser aceito com os comentários vermelhos.
</div>

Você pode me responder usando isto:

<div class="alert alert-block alert-info">
<b>Resposta do aluno.  
    </b> <a class="tocSkip"></a>
</div>



<div class="alert alert-block alert-success">

<b>Comentário geral: </b> <a class="tocSkip"></a>

    
Parabéns pela conclusão do projeto, Lucas! 🎉 Seu projeto ficou muito bom e todas as tarefas foram realizadas corretamente! 

O projeto está aprovado. Parabéns!
</div>

# Se liga na música

# Conteúdo <a id='back'></a>

* [Introdução](#intro)
* [Etapa 1. Visão geral dos dados](#data_review)
    * [Conclusões](#data_review_conclusions)
* [Etapa 2. Pré-processamento de dados](#data_preprocessing)
    * [2.1 Estilo do cabeçalho](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusões](#data_preprocessing_conclusions)
* [Etapa 3. Teste da hipótese](#hypothesis)
    * [3.1 Hipótese 1: atividade dos usuários nas duas cidades](#activity)
* [Conclusões](#end)

## Introdução <a id='intro'></a>
O trabalho de um analista é analisar dados para obter percepções valiosas dos dados e tomar decisões fundamentadas neles. Esse processo consiste em várias etapas, como visão geral dos dados, pré-processamento dos dados e testes de hipóteses.

Sempre que fazemos uma pesquisa, precisamos formular uma hipótese que depois poderemos testar. Às vezes nós aceitamos essas hipóteses; outras vezes, nós as rejeitamos. Para fazer as escolhas certas, um negócio deve ser capaz de entender se está fazendo as suposições certas ou não.

Neste projeto, você vai comparar as preferências musicais dos habitantes de Springfild e Shelbyville. Você vai estudar os dados de um serviço de streaming de música online para testar a hipótese apresentada abaixo e comparar o comportamento dos usuários dessas duas cidades.

### Objetivo:
Teste a hipótese:
1. A atividade dos usuários é diferente dependendo do dia da semana e da cidade.


### Etapas
Os dados sobre o comportamento do usuário são armazenados no arquivo `/datasets/music_project_en.csv`. Não há informações sobre a qualidade dos dados, então será necessário examiná-los antes de testar a hipótese.

Primeiro, você avaliará a qualidade dos dados e verá se seus problemas são significativos. Depois, durante o pré-processamento dos dados, você tentará tratar dos problemas mais críticos.

O seu projeto consistirá em três etapas:
 1. Visão geral dos dados
 2. Pré-processamento de dados
 3. Teste da hipótese








[Voltar ao Índice](#back)

## Etapa 1. Visão geral dos dados <a id='data_review'></a>

Abra os dados e examine-os.

Você precisará da `pandas`, então, importe-a.

In [44]:
import pandas as pd# importando pandas


Leia o arquivo `music_project_en.csv` da pasta `/datasets/` e salve-o na variável `df`:

In [45]:
df = pd.read_csv('/datasets/music_project_en.csv') # lendo o arquivo e armazenando em df


Imprima as primeiras 10 linhas da tabela:

In [46]:
print(df.head(10)) # obtenha as 10 primeiras 10 linhas da tabela df


     userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5

Obtenha informações gerais sobre a tabela usando um comando. Você conhece o método para exibir informações gerais que precisamos obter.

In [47]:
print(df.info) # obtendo informações gerais sobre os nossos dados


<bound method DataFrame.info of          userID                              Track            artist  \
0      FFB692EC                  Kamigata To Boots  The Mass Missile   
1      55204538        Delayed Because of Accident  Andreas Rönnberg   
2        20EC38                  Funiculì funiculà       Mario Lanza   
3      A3DD03C9              Dragons in the Sunset        Fire + Ice   
4      E2DC1FAE                        Soul People        Space Echo   
...         ...                                ...               ...   
65074  729CBB09                            My Name            McLean   
65075  D08D4A55  Maybe One Day (feat. Black Spade)       Blu & Exile   
65076  C5E3A0D5                          Jalopiina               NaN   
65077  321D0506                      Freight Train     Chas McDevitt   
65078  3A64EF84          Tell Me Sweet Little Lies      Monica Lopez   

            genre       City        time        Day  
0            rock  Shelbyville  20:28:33  Wednesd

Aqui estão as nossas observações sobre a tabela. Ela contém sete colunas. Elas armazenam o mesmo tipo de dado: `object`.

De acordo com a documentação:
- `' userID'` — identificação do usuário
- `'Track'` — título da música
- `'artist'` — nome do artista
- `'genre'` — gênero da música
- `'City'` — cidade do usuário
- `'time'` — o tempo exato que a música foi reproduzida
- `'Day'` — dia da semana

Podemos ver três problemas de estilo nos cabeçalhos da tabela:
1. Alguns cabeçalhos são escritos em letras maiúsculas, outros estão em minúsculas.
2. Alguns cabeçalhos contêm espaços.
3. `Não utilizaram o snake_case e time, poderia estar em datetime.time`.




### Escreva suas observações. Aqui estão algumas perguntas que podem ajudar: <a id='data_review_conclusions'></a>

`1.   Que tipo de dados temos nas linhas? E como podemos entender as colunas?`
 São dados "mistos" onde cada linha representa uma informação de um usuario. As colunas são referentes a cada caracteristica da tabela

`2.   Esses dados são suficientes para responder à nossa hipótese ou precisamos de mais dados?`
 Não precisamos de mais dados.
`3.   Você notou algum problema nos dados, como valores ausentes, duplicados ou tipos de dados errados`
 Sim, logo no Indice 9 temos um valor "Nan" e o tipo de dados da coluna time, poderia ser datetime ao invés de Object.


<div class="alert alert-block alert-success">
<b>Comentário: </b> <a class="tocSkip"></a>

Concordo com as respostas do aluno.


</div>

[Voltar ao Índice](#back)

## Etapa 2. Pré-processamento de dados <a id='data_preprocessing'></a>

O objetivo aqui é preparar os dados para a análise.
O primeiro passo é resolver todos os problemas com o cabeçalho. E então podemos passar para os valores ausentes e duplicados. Vamos começar.

Corrija a formatação nos cabeçalhos da tabela.


### Estilo do cabeçalho <a id='header_style'></a>
Imprima os cabeçalhos da tabela (os nomes das colunas):

In [48]:
# imprima os nomes das colunas
print(df.columns)

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')


Mude os cabeçalhos da tabela conforme as boas práticas de estilo:
* Todos os caracteres precisam estar com letras minúsculas
* Exclua espaços
* Se o nome tiver várias palavras, use snake_case

Anteriormente, você aprendeu sobre uma maneira automatizada de renomear colunas. Vamos usá-la agora. Use o ciclo for para percorrer os nomes das colunas e transformar todos os caracteres em letras minúsculas. Após fazer isso, imprima os cabeçalhos da tabela novamente:

In [49]:
 # Percorrendo os cabeçalhos e convertendo tudo em minúsculos

#optei por fazer junto na mesma célula
new_col_names=[]

for old_name in df.columns:
    name_lowered = old_name.lower()
    name_stripped = name_lowered.strip()
    new_col_names.append(name_stripped)
    
df.columns= new_col_names


Agora, usando a mesma abordagem, exclua os espaços no início e no final de cada nome de coluna e imprima os nomes das colunas novamente:

In [50]:
# Percorrendo os cabeçalhos e removendo os espaços

# optei por fazer junto na célula anterior
new_col_names=[]

for old_name in df.columns:
    name_lowered = old_name.lower()
    name_stripped = name_lowered.strip()
    new_col_names.append(name_stripped)
    
df.columns= new_col_names


Precisamos aplicar a regra de sublinhado no lugar de espaço à coluna `userid`. Deveria ser `user_id`. Renomeie essa coluna e imprima os nomes de todas as colunas quando terminar.

In [51]:
df= df.rename(columns={'userid':'user_id'})# Renomeando a coluna "userid"


Verifique o resultado. Imprima os cabeçalhos novamente:

In [52]:
print(df.columns) # verificando o resultado: a lista de cabeçalhos


Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


[Voltar ao Índice](#back)

### Valores Ausentes <a id='missing_values'></a>
 Primeiro, encontre a quantidade de valores ausentes na tabela. Você precisa usar dois métodos em sequência para obter o número de valores ausentes.

In [53]:
miss_value = df.isna().sum()
print(miss_value)
# isna para valores ausentes e sum para somar em cada tabela. variavel miss_value como boas praticas.


user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64


Nem todos os valores ausentes afetam a pesquisa. Por exemplo, os valores ausentes em `track` e `artist` não são críticos. Você pode simplesmente substituí-los por valores padrão, como a string `'unknown'`.

Mas valores ausentes em `'genre'` podem afetar a comparação de preferências musicais de Springfield e Shelbyville. Na vida real, seria útil descobrir as razões pelas quais os dados estão ausentes e tentar corrigi-los. Mas nós não temos essa possibilidade neste projeto. Então, você terá que:
* Preencha esses valores ausentes com um valor padrão
* Avalie em que medida os valores ausentes podem afetar sua análise

Substitua os valores ausentes nas colunas `'track'`, `'artist'` e `'genre'` pela string `'unknown'`. Como mostramos nas lições anteriores, a melhor maneira de fazer isso é criar uma lista para armazenar os nomes das colunas nas quais precisamos fazer a substituição. Em seguida, use essa lista e percorra as colunas nas quais a substituição seja necessária e faça a substituição.

In [54]:
columns_to_replace=['track', 'artist', 'genre']

for col in columns_to_replace:
    df[col].fillna('unknown', inplace=True) # percorrendo os cabeçalhos e substituindo valores ausentes por 'unknown'


Agora verifique o resultado para ter certeza de que o conjunto de dados não contenha valores ausentes após a substituição. Para fazer isso, conte os valores ausentes novamente.

In [55]:
miss_value = df.isna().sum()
print(miss_value) # contando os valores ausentes


user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64


[Voltar ao Índice](#back)

### Duplicados <a id='duplicates'></a>
Encontre o número de duplicados explícitos na tabela. Lembre-se de que você precisa aplicar dois métodos em sequência para obter o número de duplicados explícitos.

In [56]:
# contando duplicados explícitos

duplicatas = df.duplicated().sum()
print('número de duplicatas:', duplicatas)


número de duplicatas: 3826


Agora descarte todos os duplicados. Para fazer isso, chame o método que faz exatamente isso.

In [57]:
# removendo duplicados explícitos
# usei reset_index para reorganizar os indices.
df = df.drop_duplicates().reset_index(drop=True)


Agora vamos verificar se descartamos todos os duplicados. Conte duplicados explícitos mais uma vez para ter certeza de que você removeu todos eles:

In [58]:
# verificando duplicados novamente
duplicatas = df.duplicated().sum()
print('número de duplicatas:', duplicatas)




número de duplicatas: 0


Agora queremos nos livrar dos duplicados implícitos na coluna `genre`. Por exemplo, o nome de um gênero pode ser escrito de maneiras diferentes. Alguns erros afetarão também o resultado.

Para fazer isso, vamos começar imprimindo uma lista de nomes de gênero únicos, ordenados em ordem alfabética: Para fazer isso:
* Extraia a coluna `genre` do DataFrame
* Chame o método que retornará todos os valores únicos na coluna extraída


In [59]:
genre_music = sorted(df['genre'].unique(), key=str.lower) #trouxe com ordem alfabética para poder identificar possiveis erros
n_genre_musica =df['genre'].nunique() # visualizando nomes de gêneros únicos

print(genre_music)
print(n_genre_musica)

['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

Olhe a lista e encontre duplicados implícitos do gênero `hiphop`. Esses podem ser nomes escritos incorretamente, ou nomes alternativos para o mesmo gênero.

Você verá os seguintes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para se livrar deles, crie uma função `replace_wrong_genres()` com dois parâmetros:
* `wrong_genres=` — essa é uma lista que contém todos os valores que você precisa substituir
* `correct_genre=` — essa é uma string que você vai usar para a substituição

Como resultado, a função deve corrigir os nomes na coluna `'genre'` da tabela `df`, isto é, substituindo cada valor da lista `wrong_genres` por valores de `correct_genre`.

Dentro do corpo da função, use um ciclo `'for'` para percorrer a lista de gêneros errados, extrair a coluna `'genre'` e aplicar o método `replace` para fazer as correções.

In [60]:
 #alterando valores conforme enunciado

def replace_wrong_genres(df, column, wrong_genres, correct_genre): # função para automatizar substituir duplicados implícitos
    for wrong_genre in wrong_genres:
        df[column]=df[column].replace(wrong_genre,correct_genre)
    return df



Agora, chame a função `replace_wrong_genres()` e passe argumentos apropriados para que ela limpe duplicados implícitos (`hip`, `hop` e `hip-hop`) substituindo-os por `hiphop`:

In [61]:
df = replace_wrong_genres(df, 'genre', ['hip', 'hop', 'hip-hop'], 'hiphop')# removendo duplicados implícitos


Certifique-se que os nomes duplicados foram removidos. Imprima a lista de valores únicos da coluna `'genre'` mais uma vez:

In [62]:
genre_music = sorted(df['genre'].unique(), key=str.lower) #trouxe com ordem alfabética para poder identificar possiveis erros
n_genre_musica =df['genre'].nunique() # visualizando nomes de gêneros únicos

print(genre_music)
print(n_genre_musica)


['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

[Voltar ao Índice](#back)

### Suas observações <a id='data_preprocessing_conclusions'></a>

` Descreva brevemente o que você reparou ao analisar duplicados, bem como a abordagem que usou para eliminá-los e os resultados que alcançou.`
foram removidos 3 duplicados que estavam com a descrção do genero errada, antes haviam 269 e agora ha 266 generos. utilizado a função para automatizar e logo após declarado o que eu queria alterar

<div class="alert alert-block alert-success">
<b>Comentário: </b> <a class="tocSkip"></a>

Todas as etapas de processamento realizadas com sucesso.


</div>

[Voltar ao Índice](#back)

## Etapa 3. Teste da hipótese <a id='hypothesis'></a>

### Hipótese: comparação do comportamento dos usuários nas duas cidades <a id='activity'></a>

A hipótese afirma que existem diferenças no consumo de música pelos usuários em Springfield e em Shelbyville. Para testar a hipótese, use os dados dos três dias da semana: segunda-feira (Monday), quarta-feira (Wednesday) e sexta-feira (Friday).

* Agrupe os usuários por cidade.
* Compare o número de músicas tocadas por cada grupo na segunda, quarta e sexta.


Execute cada cálculo separadamente.

O primeiro passo é avaliar a atividade dos usuários em cada cidade. Não se esqueça das etapas "divisão-aplicação-combinação" sobre as quais falamos anteriormente na lição. Agora seu objetivo é agrupar os dados por cidade, aplicar o método de contagem apropriado durante a etapa de aplicação e então encontrar o número de músicas tocadas por cada grupo, especificando a coluna para a qual você quer obter a contagem.

Veja um exemplo de como o resultado final deve ser:
`df.groupby(by='....')['column'].method()` Execute cada cálculo separadamente.

Para avaliar a atividade dos usuários em cada cidade, agrupe os dados por cidade e encontre o número de músicas reproduzidas em cada grupo.



In [63]:
count_per_city = df.groupby(by='city')['user_id'].count()
print(count_per_city) # Contando as músicas tocadas em cada cidade


city
Shelbyville    18512
Springfield    42741
Name: user_id, dtype: int64


trouxeu o numero de musicas escutadas por cidade, (by=city) para o código ficar legivél. e podemos observar que Springfield ouve musicas mais que o dobro de Shelbyville.
poderiamos fazer também uma analise baseada no numero de habitantes para cada cidade.

Agora vamos agrupar os dados por dia da semana e encontrar a quantidade de músicas tocadas na segunda, quarta e sexta-feira. Use a mesma abordagem que antes, mas agora precisamos agrupar os dados de uma forma diferente.


In [64]:
count_per_day = df.groupby(by='day')['user_id'].count()
print(count_per_day)# Calculando as músicas escutadas em cada um desses três dias


day
Friday       21840
Monday       21354
Wednesday    18059
Name: user_id, dtype: int64


trouxe o numero de musicas escutadas por dias da semana, podemos ver um aumento de consumo conforme as semana se aproxima do final de semana.

Você acabou de aprender como contar entradas agrupando-as por cidade ou por dia. E agora você precisa escrever uma função que possa contar entradas simultaneamente com base em ambos os critérios.

Crie a função `number_tracks()` para calcular o número de músicas tocadas em um determinado dia **e** em uma determinada cidade. A função deve aceitar dois parâmetros:

- `day`: um dia da semana pelo qual precisamos filtrar os dados. Por exemplo, `'Monday'`.
- `city`: uma cidade pela qual precisamos filtrar os dados. Por exemplo, `'Springfield'`.

Dentro da função, você vai aplicar uma filtragem consecutiva com indexação lógica.

Primeiro, filtre os dados por dia e então filtre a tabela resultante por cidade.

Depois de filtrar os dados usando os dois critérios, conte o número de valores na coluna 'user_id' da tabela resultante. O resultado da contagem representará o número de entradas que você quer encontrar. Armazene o resultado em uma nova variável e imprima-o.

In [65]:
def number_tracks(day,city):# Declare a função number_tracks() com dois parâmetros: day= e city=.

    filtred_df = df[(df['day'] == day) & (df['city'] == city)]# Armazene as linhas do DataFrame em que o valor na coluna 'day' é igual ao parâmetro day= # Filtre as linhas em que o valor na coluna 'city' é igual ao parâmetro city=

    total_reproduces = filtred_df['user_id'].count()   # Extraia a coluna 'user_id' da tabela filtrada e aplique o método count()

    return total_reproduces # Retorne o número dos valores da coluna 'user_id'

Chame a função `number_tracks()` seis vezes, mudando os valores dos parâmetros, para que você possa recuperar os dados de ambas as cidades para cada um dos três dias.

In [66]:
springfield_monday = number_tracks('Monday', 'Springfield')
print(Springfield_monday)# a quantidade de músicas tocadas em Springfield na segunda-feira


16715


In [67]:
shelbyville_monday = number_tracks('Monday', 'Shelbyville')
print(shelbyville_monday)# a quantidade de músicas tocadas em Shelbyville na segunda-feira


5614


In [68]:
springfield_wednesday = number_tracks('Wednesday', 'Springfield')
print(springfield_wednesday)# a quantidade de músicas tocadas em Springfield na quarta-feira


11056


In [69]:
shelbyville_wednesday = number_tracks('Wednesday', 'Shelbyville')
print(shelbyville_wednesday)# a quantidade de músicas tocadas em Shelbyville na quarta-feira


7003


In [70]:
springfield_friday = number_tracks('Friday', 'Springfield')
print(springfield_friday)# a quantidade de músicas tocadas em Springfield na sexta-feira


15945


In [71]:
shelbyville_friday = number_tracks('Friday', 'Shelbyville')
print(shelbyville_friday)# a quantidade de músicas tocadas em Shelbyville na sexta-feira


5895


**Conclusões**

`Comente sobre se a terceira hipótese está correta ou deve ser rejeitada. Explique seu raciocínio.`
A hipótese que afirma que existem diferenças no consumo de música pelos usuários em Springfield e em Shelbyville esta totalmente correta, não apenas parcialmente. Visto que o numero de musicas ouvidas em Springfield (42741) é mais que o dobro de Shelbyville

<div class="alert alert-block alert-success">
<b>Comentário: </b> <a class="tocSkip"></a>

- Ambas as funções foram declaradas com sucesso. Parabéns!
- Foram adicionados comentários relevantes pra análise



</div> 

[Voltar ao Índice](#back)

# Conclusões <a id='end'></a>

`Resuma suas conclusões sobre a hipótese aqui`
Foram feitos as etapas de agrupamento (dividir,aplicar,combinar) para testar as hipoteses. Se tivessemos os habitantes de cada cidade também poderiamos trazer a porcetagem de ouvintes em cada cidade, onde baseado na população poderiamos trazer qual cidade "consome" musicas.

### Importante
Em projetos de pesquisas reais, o teste estatístico de hipóteses é mais preciso e quantitativo. Observe também que conclusões sobre uma cidade inteira nem sempre podem ser tiradas a partir de dados de apenas uma fonte.

Você aprenderá mais sobre testes de hipóteses no sprint sobre a análise estatística de dados.

[Voltar ao Índice](#back)